libcore/test/catch2/geometry/GeometryHelperTest.cpp

```diff
@@ -26,580 +26,998 @@
+
+                distance = distance + width + 0.25 * trackWall2.GetLength();
+                width    = 0.25 * trackWall2.GetLength() + 0.25 * trackWall3.GetLength();
+                trainDoors.emplace_back(TrainDoor{distance, width, outflow});
+
+                distance = distance + width + 0.5 * trackWall3.GetLength();
+                width    = 0.25 * trackWall3.GetLength() + trackWall4.GetLength() +
+                        0.25 * trackWall5.GetLength();
+                trainDoors.emplace_back(TrainDoor{distance, width, outflow});
+
+                TrainType train{"TEST", 20, 10, trainDoors};
+
+                auto doorCoordinates = geometry::helper::ComputeTrainDoorCoordinates(
+                    train, track, trainStartOffSet, false);
+
+                REQUIRE(doorCoordinates.size() == trainDoors.size());
+                for(size_t i = 0; i < doorCoordinates.size(); ++i) {
+                    // check distance to start
+                    auto doorCoordinate   = doorCoordinates[i];
+                    auto wallDoorStartItr = std::find_if(
+                        std::begin(trackWalls),
+                        std::end(trackWalls),
+                        [&doorCoordinate](const Wall & wall) {
+                            return wall.IsInLineSegment(doorCoordinate.first);
+                        });
+                    double distanceIs = std::accumulate(
+                        std::begin(trackWalls),
+                        wallDoorStartItr,
+                        0.,
+                        [](double & sum, const Wall & wall) { return sum + wall.GetLength(); });
+                    distanceIs += Distance(wallDoorStartItr->GetPoint1(), doorCoordinate.first);
+                    double distanceShould = trainDoors[i]._distance + trainStartOffSet;
+                    REQUIRE(distanceIs == Approx(distanceShould));
+
+                    // check distance of door elements
+                    REQUIRE(
+                        trainDoors[i]._width ==
+                        Approx(Distance(doorCoordinate.first, doorCoordinate.second)));
+
+                    // check if on any of these lines
+                    auto wallPointItr = std::find_if(
+                        std::begin(trackWalls),
+                        std::end(trackWalls),
+                        [&doorCoordinate](const Wall & wall) {
+                            return wall.IsInLineSegment(doorCoordinate.first);
+                        });
+                    REQUIRE_FALSE(wallPointItr == std::end(trackWalls));
+
+                    wallPointItr = std::find_if(
+                        std::begin(trackWalls),
+                        std::end(trackWalls),
+                        [&doorCoordinate](const Wall & wall) {
+                            return wall.IsInLineSegment(doorCoordinate.second);
+                        });
+                    REQUIRE_FALSE(wallPointItr == std::end(trackWalls));
+                }
```

Comment:
Did you test a door which is defined outside a valid track wall? (train longer than track)

Reply:
Added missing tests

Reply:
What happens in the simulation? transitions are only created for doors on track walls, right?

Reply:
yes, every door outside of the track walls will be ignored